### PR TITLE
Terms: Replace callback based functions with suspend functions

### PR DESCRIPTION
**Pull request by markusrubey**

### CODE REVIEW

File: demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt [hunk 1]
```diff
@@ -23,51 +28,57 @@ class TermsViewModel : ViewModel() {
...
+                viewStateInternal.value = viewStateInternal.value?.copy(
+                        isLoading = false,
+                        errorMessage = result.error.toString()
+                )
+            }
+        }
     }
 
-    fun acceptTerms() {
+    fun acceptTerms() = viewModelScope.launch {
```
Comment (lucassales2) [inline demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt]: Although this is a better style it exposes the coroutine job to the view, giving it the power to cancel it

Comment (markusrubey) [reply demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt]: agree, see 
09f2049

File: demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt [hunk 2]
```diff
@@ -24,50 +29,60 @@ class TermsViewModel : ViewModel() {
...
+                )
+            }) {
+                is Result.Success -> {
                     viewStateInternal.value = viewStateInternal.value?.copy(
-                            isLoading = false,
-                            errorMessage = null,
-                            isAccepted = true
+                        isLoading = false,
+                        errorMessage = null,
+                        isAccepted = true
```
Comment (lucassales2) [inline demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt]: apply spotless so we keep the same code style

Comment (markusrubey) [reply demo/src/main/java/dk/nodes/nstack/demo/terms/TermsViewModel.kt]: fixed in 96ed984

**Pull request merged**